samples/samples/admin/backup_snippet_test.py

```diff
@@ -0,0 +1,191 @@
+@RetryErrors(exception=DeadlineExceeded, max_tries=2)
+def test_restore_database_with_encryption_key(
+    capsys,
+    instance_id,
+    sample_database,
+    kms_key_name,
+):
+    backup_snippet.restore_database_with_encryption_key(
+        instance_id, CMEK_RESTORE_DB_ID, CMEK_BACKUP_ID, kms_key_name
+    )
+    out, _ = capsys.readouterr()
+    assert (sample_database.database_id + " restored to ") in out
+    assert (CMEK_RESTORE_DB_ID + " from backup ") in out
+    assert CMEK_BACKUP_ID in out
+    assert kms_key_name in out
+
+
+@pytest.mark.dependency(depends=["create_backup", "copy_backup"])
+def test_list_backup_operations(capsys, instance_id, sample_database):
+    backup_snippet.list_backup_operations(
+        instance_id, sample_database.database_id, BACKUP_ID
+    )
+    out, _ = capsys.readouterr()
+    assert BACKUP_ID in out
+    assert sample_database.database_id in out
+    assert COPY_BACKUP_ID in out
+    print(out)
+
+
+@pytest.mark.dependency(name="list_backup", depends=["create_backup", "copy_backup"])
+def test_list_backups(
+    capsys,
+    instance_id,
+    sample_database,
+):
+    backup_snippet.list_backups(
+        instance_id,
+        sample_database.database_id,
+        BACKUP_ID,
+    )
+    out, _ = capsys.readouterr()
+    id_count = out.count(BACKUP_ID)
+    assert id_count == 7
+
+
+@pytest.mark.dependency(depends=["create_backup"])
+def test_update_backup(capsys, instance_id):
+    backup_snippet.update_backup(instance_id, BACKUP_ID)
+    out, _ = capsys.readouterr()
+    assert BACKUP_ID in out
+
+
+@pytest.mark.dependency(depends=["create_backup", "copy_backup", "list_backup"])
+def test_delete_backup(capsys, instance_id):
+    backup_snippet.delete_backup(instance_id, BACKUP_ID)
+    out, _ = capsys.readouterr()
+    assert BACKUP_ID in out
+    backup_snippet.delete_backup(instance_id, COPY_BACKUP_ID)
+    out, _ = capsys.readouterr()
+    assert COPY_BACKUP_ID in out
+
+
+@pytest.mark.dependency(depends=["create_backup"])
+def test_cancel_backup(capsys, instance_id, sample_database):
+    backup_snippet.cancel_backup(
+        instance_id,
+        sample_database.database_id,
+        BACKUP_ID,
+    )
+    out, _ = capsys.readouterr()
+    cancel_success = "Backup creation was successfully cancelled." in out
+    cancel_failure = ("Backup was created before the cancel completed." in out) and (
+        "Backup deleted." in out
+    )
+    assert cancel_success or cancel_failure
+
+
+@RetryErrors(exception=DeadlineExceeded, max_tries=2)
+def test_create_database_with_retention_period(capsys, sample_instance):
+    backup_snippet.create_database_with_version_retention_period(
+        sample_instance.instance_id,
+        RETENTION_DATABASE_ID,
+        RETENTION_PERIOD,
+    )
+    out, _ = capsys.readouterr()
+    assert (RETENTION_DATABASE_ID + " created with ") in out
+    assert ("retention period " + RETENTION_PERIOD) in out
```

Comment:
make sure this database is dropped

Reply:
We are deleting the database in the sample itself.